docs/contribution.md

```diff
@@ -94,43 +94,63 @@ All submodule should have its own pytest section under `tests/units/anta_tests/<
 
-The Python modules in the `tests/units/anta_tests` folder  define test parameters for AntaTest subclasses unit tests.
-A generic test function is written for all unit tests in `tests.units.anta_tests` module.
+The Python modules in the `tests.units.anta_tests` package define test parameters for AntaTest subclasses unit tests.
+A generic test function is written for all unit tests of the `AntaTest` subclasses.
+In order for your unit tests to be correctly collected, you need to import the generic test function even if not used in the Python module.
 
 The `pytest_generate_tests` function definition in `conftest.py` is called during test collection.
 
-The `pytest_generate_tests` function will parametrize the generic test function based on the `DATA` data structure defined in `tests.units.anta_tests` modules.
+The `pytest_generate_tests` function will parametrize the generic test function based on the `DATA` constant defined in modules in the `tests.units.anta_tests` package.
 
 See https://docs.pytest.org/en/7.3.x/how-to/parametrize.html#basic-pytest-generate-tests-example
 
-The `DATA` structure is a list of dictionaries used to parametrize the test. The list elements have the following keys:
+The `DATA` constant is a list of dictionaries used to parametrize the test. A `TypedDict` has been defined to ease the writing of such constant and leverage `mypy` type checking.
 
-- `name` (str): Test name as displayed by Pytest.
-- `test` (AntaTest): An AntaTest subclass imported in the test module - e.g. VerifyUptime.
-- `eos_data` (list[dict]): List of data mocking EOS returned data to be passed to the test.
-- `inputs` (dict): Dictionary to instantiate the `test` inputs as defined in the class from `test`.
-- `expected` (dict): Expected test result structure, a dictionary containing a key
-    `result` containing one of the allowed status (`Literal['success', 'failure', 'unset', 'skipped', 'error']`) and optionally a key `messages` which is a list(str) and each message is expected to  be a substring of one of the actual messages in the TestResult object.
+``` python
+class AtomicResult(TypedDict):
+    """Expected atomic result of a unit test of an AntaTest subclass."""
 
-In order for your unit tests to be correctly collected, you need to import the generic test function even if not used in the Python module.
+    result: Literal["success", "failure", "skipped"]  # The expected status of this atomic result.
+    description: str  # The expected description of this atomic result.
+    messages: NotRequired[list[str]]  # The expected messages of this atomic result. The strings can be a substrings of the actual messages.
+    inputs: NotRequired[dict[str, Any]]  # The inputs registered with this atomic result.
+
+
+class Expected(TypedDict):
+    """Expected result of a unit test of an AntaTest subclass."""
+
+    result: Literal["success", "failure", "skipped"]  # The expected status of this unit test.
```

Comment:
should it not have more (unset / error)? or is it used only in places where we expect these 3?

Reply:
No unit test case expects unset (test has never run) or error (an unexpected error has been caught by the ANTA framework, not by the test code).